sdk/core/azure-core-test/src/test_proxy_manager.cpp

```diff
@@ -242,7 +242,7 @@ void TestProxyManager::SetProxySanitizer()
 
   addSanitizer(SanitizerType::General, g_accountRegex, "account");
   addSanitizer(
-      SanitizerType::Body, "client_secret=(?<clientsecret>[a-zA-Z0-9\\%]+)", "clientsecret");
+      SanitizerType::Body, "client_secret=(?<clientsecret>[a-zA-Z0-9\\%_~]+)", "clientsecret");
```

Comment:
```suggestion
      SanitizerType::Body, "client_secret=(?<clientsecret>[a-zA-Z0-9\\%_~\-\.]+)", "clientsecret");
```
This should fix it once and for all.

Explanation:
https://github.com/Azure/azure-sdk-for-cpp/blob/15d8133a215a1f1d93c40714455e9f4e8afb4013/sdk/identity/azure-identity/src/client_secret_credential.cpp#L32

=>

https://github.com/Azure/azure-sdk-for-cpp/blob/15d8133a215a1f1d93c40714455e9f4e8afb4013/sdk/core/azure-core/src/http/url.cpp#L134

And, I just made Azure Portal to generate 5 random AAD secrets for me, some of them had a dash and a dot.